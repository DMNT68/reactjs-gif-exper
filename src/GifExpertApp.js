import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';

export const GifExpertApp = ({ defaultCategories = [] }) => {
  //   const categories = ['One Punchman', 'Samurai X', 'Dragon ball'];
  // const [categories, setCategories] = useState(['One puchman']);
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <div>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={setCategories} />
      <hr />

      <ol>
        {categories.map((category, idx) => (
          <GifGrid key={`${(category = idx)}`} category={category} />
        ))}
      </ol>
    </div>
  );
};
