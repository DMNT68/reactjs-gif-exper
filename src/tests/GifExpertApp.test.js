import React from 'react';
import { shallow } from 'enzyme';
import { GifExpertApp } from '../GifExpertApp';

describe('Pruebas en <GifExpertApp/>', () => {
  test('debe mostrar correctamente el componente', () => {

    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  test('debe de mostrar una lista categorias', () => {
    const categories = ['One Punch', 'Dragon ball'];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length);
  });

});
