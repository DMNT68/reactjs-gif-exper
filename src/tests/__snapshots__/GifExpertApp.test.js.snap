// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp/> debe de mostrar una lista categorias 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Punch"
      key="0"
    />
    <GifGrid
      category="Dragon ball"
      key="1"
    />
  </ol>
</div>
`;

exports[`Pruebas en <GifExpertApp/> debe mostrar correctamente el componente 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
